### DISCUSSION TITLE
CameraView iOS; Can I/How do I engage "macro mode"?

### DISCUSSION THREAD

**Discussion by DwightOakey** [CommunityToolkit/Maui#2197] [General]
Since iOS 15.1, and with iPhone 13 or better, the camera has been able to view in "macro mode" (sometimes called wide-angle, or ultra-wide) to view clear close-ups. When using CommunityToolkit.CameraView for the last 2 years, my iOS app has not had this clear focus at close-up distances. I am now converting my old Xamarin.Forms app to .NET MAUI, and this close-up disadvantage is even more apparent. I can tell by comparing my app to the built-in Camera app that CameraView is using the nominal lens, and not the wide-angle lens, and therefore I can't "turn on" this macro mode focus.  Is there any way to turn on macro mode in CameraView?  Thanks in advance.

**Comment by bijington** [comment]
At present I don't think you can do this with the CameraView out of the box. 

It is possible on iOS using something like this: https://stackoverflow.com/a/61159532 (thanks to @brminnick for the article) so that shows it will be possible with CameraView. The tricky part comes with making something like this fit across all platforms or how we expose just the iOS bits where necessary.

Long story short it will be possible when someone implements it. If this is something you are keen to use would you be willing to implement the changes required to support your scenario for us?

**Comment by DwightOakey** [comment]
Willing? Perhaps. Capable? I'm afraid not.

For me, this is a shame. I have an OCR app that uses the camera to convert model & serial numbers into searchable text. Before iOS 15.1, I could use my fingers to steady the shot and get clear images of my source.  (NOTE: I can still do this on Android, with this same app.)  However, since iOS 15.1, I have to hold my hands about 6-7 inches (approx. 150-170mm) away from my target. This makes it much harder to hold steady to get a clear shot. If we could tap into "macro mode", I could go back to a hold distance of 3 inches or less (under 70mm). 

Macro mode in the Apple Camera app is the little yellow flower:
![image](https://github.com/user-attachments/assets/660c7a0a-f298-478e-b204-168f4b9f75d5)

Thanks.

**Reply by stephenquan** [reply]
Perhaps this focusing trick may work for you. I put the phone dead ontop of the target, then, slowly pull the phone backwards. This seems to reset the autofocus for me. Sometimes it may take 2x or 3x attempts to do make it work.
